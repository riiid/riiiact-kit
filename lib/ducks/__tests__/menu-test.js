import reducer, {
  requestMenu, receiveMenuSuccess, receiveMenuFailure
} from 'ducks/menu';

describe('actions/menu', () => {
  it('should handle requestMenu()', () => {
    expect(reducer(undefined, requestMenu())).toMatchSnapshot();
  });

  it('should receiveMenuSuccess', () => {
    expect(reducer(undefined, receiveMenuSuccess({
      currentViews: 0,
      totalViews: 0
    }))).toMatchSnapshot();
  });

  it('should receiveMenuFailure', () => {
    expect(reducer(undefined, receiveMenuFailure({}))).toMatchSnapshot();
  });
});
