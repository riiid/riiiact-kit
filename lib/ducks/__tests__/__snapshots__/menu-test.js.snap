// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions/menu should handle requestMenu() 1`] = `
Object {
  "count": Object {
    "currentViews": 0,
    "totalViews": 0,
  },
}
`;

exports[`actions/menu should receiveMenuFailure 1`] = `
Object {
  "count": Object {
    "currentViews": "N/A",
    "totalViews": "N/A",
  },
  "error": Object {},
}
`;

exports[`actions/menu should receiveMenuSuccess 1`] = `
Object {
  "count": Object {
    "currentViews": 0,
    "totalViews": 0,
  },
  "error": null,
}
`;
